crawl-ref/source/beam.cc

```diff
@@ -3237,15 +3230,9 @@ bool bolt::misses_player()
     else
     {
         int dodge_more = you.evasion(EV_IGNORE_HELPLESS);
-
-        if (hit_verb.empty())
-            hit_verb = engulfs ? "engulfs" : "hits";
-
-        if (_test_beam_hit(real_tohit, dodge_more, pierce, defl, r))
-            mprf("The %s %s you!", name.c_str(), hit_verb.c_str());
-        else
+        if (!_test_beam_hit(real_tohit, dodge_more, pierce, defl, r))
             mprf("Helpless, you fail to dodge the %s.", name.c_str());
-
+        // Other hit messages are done by the caller.
         miss = false;
```

Comment:
What would you prefer to be done about this helpless message? I currently have it so that it prints in addition to the `"The %s hits you!!!"` message, because it was necessary to move the hit messaging out of `bolt::misses_player` so that they could be printed after the bolt's damage had been calculated, but this helpless message relies on `_test_beam_hit` taking arguments that are variables local to the `bolt::misses_player` method, so it's not a simple task to move it.

Reply:
In melee there is no message for when you could have dodged had you not been helpless, so you could just remove this message. It's nice flavour but it doesn't add much to know that you could have dodge that crystal spear had you not been paralyzed. It also seems to trigger while in tree form, which is a bit odd, as the player isn't *helpless* as a tree. Just a tree.